### PR TITLE
requests: annotate `utils.get_encoding_from_headers()`

### PR DESCRIPTION
**Pull request by deadly-panda**

### CODE REVIEW

**Review by srittau — CHANGES_REQUESTED** [review]

File: stubs/requests/requests/utils.pyi [hunk 1]
```diff
@@ -34,7 +34,7 @@ def unquote_header_value(value, is_filename: bool = False): ...
 def dict_from_cookiejar(cj): ...
 def add_dict_to_cookiejar(cj, cookie_dict): ...
 def get_encodings_from_content(content): ...
-def get_encoding_from_headers(headers): ...
+def get_encoding_from_headers(headers: MutableMapping[str, str | bytes]) -> str | None: ...
```
[the remark this answers is not in the document]
Comment (deadly-panda) [reply stubs/requests/requests/utils.pyi:37]: Thanks for the review!
What about `CaseInsensitiveDict`?

[the remark this answers is not in the document]
Comment (JelleZijlstra) [reply stubs/requests/requests/utils.pyi:37]: `CaseInsensitiveDict[str]` should be a subtype of `Mapping[str, str]` (though I haven't checked), so Sebastian's suggestion should get rid of the mypy-primer hit.

[the remark this answers is not in the document]
Comment (deadly-panda) [reply stubs/requests/requests/utils.pyi]: Updated, Thanks!

**Review by srittau — APPROVED** [review]
Thanks!

**Pull request merged**